src/CalcManager/Ratpack/conv.cpp

```diff
@@ -436,15 +356,14 @@ PRAT StringToRat(bool mantissaIsNegative, wstring_view mantissa, bool exponentIs
     else
     {
         // Mantissa specified, convert to number form.
-        PNUMBER pnummant = StringToNumber(mantissa, radix, precision);
-        if (pnummant == nullptr)
+        optional<NUMBER> pnummant = StringToNumber(mantissa, radix, precision);
```

Comment:
`std::optional` works well with `auto` and pointer semantics:
```suggestion
        auto pnummant = StringToNumber(mantissa, radix, precision);
        if (!pnummant)
        {
            ...
        }
```